Tests/SwiftFusionTests/Inference/ChordalInitializationTests.swift

```diff
@@ -0,0 +1,144 @@
+// Copyright 2020 The SwiftFusion Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import TensorFlow
+import XCTest
+
+import PenguinStructures
+import SwiftFusion
+
+// MARK: sample data
+
+// symbol shorthands
+let x0 = TypedID<Pose3, Int>(0)
+let x1 = TypedID<Pose3, Int>(1)
+let x2 = TypedID<Pose3, Int>(2)
+let x3 = TypedID<Pose3, Int>(3)
+
+// ground truth
+let  p0 = Vector3(0,0,0);
+let  R0 = Rot3.fromTangent(Vector3(0.0,0.0,0.0))
+let  p1 = Vector3(1,2,0)
+let  R1 = Rot3.fromTangent(Vector3(0.0,0.0,1.570796))
+let  p2 = Vector3(0,2,0)
+let  R2 = Rot3.fromTangent(Vector3(0.0,0.0,3.141593))
+let  p3 = Vector3(-1,1,0)
+let  R3 = Rot3.fromTangent(Vector3(0.0,0.0,4.712389))
+
+let pose0 = Pose3(R0,p0)
+let pose1 = Pose3(R1,p1)
+let pose2 = Pose3(R2,p2)
+let pose3 = Pose3(R3,p3)
```

Comment:
If you don’t need the R and P just create in place. Also, try to also use the convention tRf where t=to and f=from. See the blog post on gtsam.org :-)